### PR TITLE
Add version number in header files

### LINKED ISSUE #458: Request: Add version number in header files
Please add version numbers in header files, that can be used at compile time.

For example:

```
#define RYML_VERSION "0.7.0"
#define RYML_VERSION_MAJOR 0
#define RYML_VERSION_MINOR 7
#define RYML_VERSION_PATCH 0
```

This helps when code using ryml needs to be different for v0.6.0 and v0.7.0, due to api breaking changes.

To illustrate, right now my application code looks like:

```
  ryml::ParserOptions opts;
  opts.locations(true);

#if OTEL_HAVE_RYML == 6
  ryml::Parser parser(opts);
#endif

#if OTEL_HAVE_RYML == 7
  ryml::Parser::handler_type event_handler;
  ryml::Parser parser(&event_handler, opts);
#endif

  ryml::Tree tree;
  ryml::csubstr filename;
  ryml::csubstr csubstr_content;
  std::unique_ptr<Document> doc;

  filename        = ryml::to_csubstr(source);
  csubstr_content = ryml::to_csubstr(content);

  try
  {
#if OTEL_HAVE_RYML == 6
    tree = parser.parse_in_arena(filename, csubstr_content);
#endif

#if OTEL_HAVE_RYML == 7
    tree = parse_in_arena(&parser, filename, csubstr_content);
#endif

    tree.resolve();
  }
  catch (...)
```

where `OTEL_HAVE_RYML` is evaluated by CMake, based on what find_package() actually found.

It is desirable to rely on `RYML_VERSION_MAJOR`, `RYML_VERSION_MINOR` and `RYML_VERSION_PATCH` found in the ryml header files instead.

### PR DESCRIPTION
**Pull request by marcalff**
Fixes #458

## Changes

* Add version numbers in header files, that can be used at compile time
* Add functions to get the library version numbers, that can be used at runtime. This helps with shared libraries.

### CODE REVIEW

**Comment by biojppm** [comment]
Thanks, this is appreciated!

The PR looks good, but there is one extra thing, though: add corresponding entries for the new files in the [tbump file](https://github.com/biojppm/rapidyaml/blob/52f1902fdd7c3f26cef769ab45115780b54766cf/tbump.toml#L26).

**Comment by marcalff** [comment]
> Thanks, this is appreciated!
> 
> The PR looks good, but there is one extra thing, though: add corresponding entries for the new files in the [tbump file](https://github.com/biojppm/rapidyaml/blob/52f1902fdd7c3f26cef769ab45115780b54766cf/tbump.toml#L26).

Interesting tool, will look into this.

**Comment by marcalff** [comment]
> > Thanks, this is appreciated!
> > The PR looks good, but there is one extra thing, though: add corresponding entries for the new files in the [tbump file](https://github.com/biojppm/rapidyaml/blob/52f1902fdd7c3f26cef769ab45115780b54766cf/tbump.toml#L26).
> 
> Interesting tool, will look into this.

Done

**Comment by biojppm** [comment]
Thanks! I've added some tests; let's wait for the pipelines to turn green.

**Comment by marcalff** [comment]
CI is a carnage, not sure why in most cases.

Found this for find_package()

```
/home/runner/work/rapidyaml/rapidyaml/prefix/find_package/linux-Debug/include/c4/yml/yml.hpp:4:10: fatal error: c4/yml/version.hpp: No such file or directory
```

Is there a place to update for new headers to install ?

**Comment by biojppm** [comment]
Indeed, that's missing. 

Btw, the sea of red is unrelated; it's happening because GitHub updated nodejs in the workflow actions and the ubuntu18 runner is therefore no longer supported. Let's ignore this problem. Let's focus on the workflows related to install.

File: CMakeLists.txt [hunk 1]
```diff
@@ -70,6 +70,7 @@ c4_add_library(ryml
         c4/yml/tag.cpp
         c4/yml/tree.hpp
         c4/yml/tree.cpp
+        c4/yml/version.cpp
```
Comment (marcalff) [inline CMakeLists.txt:73]:
```suggestion
        c4/yml/version.hpp
        c4/yml/version.cpp
```

**Comment by biojppm** [comment]
@marcalff warning: I've rebased your branch into the current master.

**Comment by marcalff** [comment]
> @marcalff warning: I've rebased your branch into the current master.

That is fine, I noticed all the CI cleanup required as pre requisite.

I take it this PR is to be merged after #461 is done ?

**Comment by biojppm** [comment]
> I take it this PR is to be merged after https://github.com/biojppm/rapidyaml/pull/461 is done ?

Well, by coincidence, not by necessity.

---

I've looked through the pipelines, and the relevant ones are all green (the install/release pipelines succeed, and the sample pipelines succeed for the relevant tests, failing later only because of an unrelated problem which happens with PRs of non-owners). So I'm merging now.

Thanks for your PR, and for sticking through this.

**Pull request merged**